### PR TITLE
Ban JUnit 4 from the build

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: test-framework/arquillian/pom.xml [hunk 1]
```diff
@@ -90,6 +90,27 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id >enforce</id>
+                        <configuration combine.self="append">
+                            <rules>
+                                <bannedDependencies>
+                                    <includes>
+                                        <exclude>junit:junit</exclude>
```
Comment (gsmet) [inline test-framework/arquillian/pom.xml:103]: Shouldn't it be:
```suggestion
                                        <include>junit:junit</include>
```
?

**Review by gsmet — CHANGES_REQUESTED** [review]

File: test-framework/arquillian/pom.xml (hunk 1 above)
Comment (stuartwdouglas) [reply test-framework/arquillian/pom.xml]: It should. Weirdly though it still works, and you can actually use any tag name inside the <includes> as the parser must not verify the element name.

**Comment by stuartwdouglas** [comment]
Turns out testcontainers has a hard dependency on junit 4, so this won't really work.

**Comment by famod** [comment]
That's one reason why we'd need a separate execution. For that we could say `<searchTransitive>false</searchTransitive>` and could at least make sure we have no direct dependency to it.

**Comment by famod** [comment]
Btw, the testcontainers dependency is a strange one which can be worked around by re-implementing two or three junit classes (that can be entirely empty!) 🤦 , which is what we had to do in a previous project. Not really suitable for a framework like Quarkus, I guess...

**Comment by stuartwdouglas** [comment]
Well we don't really support JUnit 4, so maybe we could have it as an option. It's really disappointing that testcontainers is so tied to JUnit 4.

**Pull request closed without merging**